master/internal/api_tasks.go

```diff
@@ -35,9 +37,68 @@ var (
 	taskNotFound = status.Error(codes.NotFound, "task not found")
 )
 
+func expFromAllocationID(
+	m *Master, allocationID model.AllocationID,
+) (isExperiment bool, exp *model.Experiment, err error) {
+	resp, err := m.rm.GetAllocationHandler(
+		m.system,
+		sproto.GetAllocationHandler{ID: allocationID},
+	)
+	if err != nil {
+		return false, nil, status.Errorf(codes.NotFound, "allocation not found: %s", allocationID)
```

Comment:
Type checking this error for allocation not found is hard since in agent rm it returns 
```errors.New("task handler not found on any resource pool")```

while in k8s it returns
```fmt.Errorf("actor %s did not response", addr)```

Are there other possible errors that can come that I missing that make this blanket ```err != nil``` return ```not found``` suppress a useful error message?

Reply:
I think a nil response ends up getting converted to an error in ```a.ask``` when a response is expected
https://github.com/determined-ai/determined/blob/48ac9af201c4ab3c8ccd56683b47c08bc35d40d6/master/internal/api.go#L198